From: Michael Biebl <biebl@debian.org>
Date: Thu, 21 Jan 2016 23:11:29 +0100
Subject: tests: use dbus-run-session instead of dbus-launch

The dbus-run-session utility was designed to run a process within a
D-Bus session, specifically for running regressions tests and is much
better suited then dbus-launch. As an additional benefit, this avoids
any X dependencies.
---
 libnm-glib/tests/libnm-glib-test-launch.sh | 6 +-----
 libnm/tests/libnm-test-launch.sh           | 6 +-----
 2 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/libnm-glib/tests/libnm-glib-test-launch.sh b/libnm-glib/tests/libnm-glib-test-launch.sh
index 980f834..1064e41 100755
--- a/libnm-glib/tests/libnm-glib-test-launch.sh
+++ b/libnm-glib/tests/libnm-glib-test-launch.sh
@@ -1,7 +1,3 @@
 #!/bin/sh
 
-# Spawn DBus
-eval `dbus-launch --sh-syntax`
-trap "kill $DBUS_SESSION_BUS_PID" EXIT
-
-"$@"
+dbus-run-session -- "$@"
diff --git a/libnm/tests/libnm-test-launch.sh b/libnm/tests/libnm-test-launch.sh
index 980f834..1064e41 100755
--- a/libnm/tests/libnm-test-launch.sh
+++ b/libnm/tests/libnm-test-launch.sh
@@ -1,7 +1,3 @@
 #!/bin/sh
 
-# Spawn DBus
-eval `dbus-launch --sh-syntax`
-trap "kill $DBUS_SESSION_BUS_PID" EXIT
-
-"$@"
+dbus-run-session -- "$@"
